### PR TITLE
Block application if already applied or if ineligible. Fix bugs.

### PR DESCRIPTION
**Pull request by DNS-404**

### CODE REVIEW

File: job.php [hunk 1]
```diff
@@ -46,97 +45,105 @@
...
+          foreach($stud_job_offers_arr as $job_offer_id){
+            if(!$is_stud_eligible){
+              break;
+            }
+            // Fetch company id using job id
+            $job_query = "SELECT company_category FROM applications WHERE job_id='" . $job_offer_id . "'";
+            $data = mysqli_query($dbc, $job_query);
+            $row = mysqli_fetch_array($data);
+            $job_offer_company_cat = $row['company_category'];
+            //echo "Offer is by" . $job_offer_company_cat . "<br/>";
```
Comment (PrP-11) [inline job.php]:
```suggestion
```

File: job.php [hunk 2]
```diff
@@ -46,97 +45,105 @@
...
+              // Able to apply for only two A1
+              // Check how many A1s has student already applied to
+              $count_A1_query = "SELECT * FROM applications WHERE student_roll_number='" . $_SESSION['roll_number'] . "' AND company_category='A1'";
+              $data = mysqli_query($dbc, $count_A1_query);
+              if(mysqli_num_rows($data) < 2){
+                // Student can apply for more A1s
+                $company_query = "SELECT company_category FROM recruiters WHERE company_id='" . $company_id . "'";
+                $data = mysqli_query($dbc, $company_query);
+                $row = mysqli_fetch_assoc($data);
+                //echo "Current company is " . $row['company_category'];
```
Comment (PrP-11) [inline job.php]:
```suggestion
```

File: job.php (hunk 1 above)
Comment (DNS-404) [reply job.php]: This is by intention. Not by mistake.

File: job.php (hunk 2 above)
Comment (DNS-404) [reply job.php]: This is by intention. Not by mistake.

**Pull request merged**